conformance/tests/httproute-path-match-order.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: path-matching-order
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - matches:
+    - path:
+        type: Exact
+        value: /match
+    backendRefs:
+    - name: infra-backend-v1
+      port: 8080
+  - matches:
+    - path:
+        type: Exact
+        value: /match/exact
+    backendRefs:
+    - name: infra-backend-v2
+      port: 8080
+  - matches:
+    - path:
+        type: Exact
+        value: /match/exact/one
+    backendRefs:
+    - name: infra-backend-v3
+      port: 8080
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /match/
+    backendRefs:
+    - name: infra-backend-v3
+      port: 8080
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /match/prefix/
+    backendRefs:
+    - name: infra-backend-v1
+      port: 8080
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /match/prefix/one/
```

Comment:
It may be helpful to have at least one of these prefix matches without a trailing slash.

Reply:
Done Thanks!